### ISSUE TITLE
False positive: unnecessary-dict-index-lookup when deleting a key in a dictionary

### ISSUE THREAD

**Issue by cpitclaudel** [PyCQA/pylint#5536]
### Bug description

Pylint suggests replacing `a[k]` below with `v`
```
# pylint: disable=missing-module-docstring
a = {"a": 1}
for k, v in a.items():
    if v % 2 == 0:
        del a[k]
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:5:12: R1733: Unnecessary dictionary index lookup, use 'v' instead (unnecessary-dict-index-lookup)
```

### Expected behavior

Ideally, an error about modifying a dictionary during iteration; alternatively, no errors.

### Pylint version

```shell
pylint 2.12.2
```

**Comment by DanielNoord** [comment]
This is a duplicate of #4716. A fix has been made in #5344 which will be released with `pylint` `2.13`!
